rtgui/batchtoolpanelcoord.cc

```diff
@@ -562,11 +563,250 @@ void BatchToolPanelCoordinator::panelChanged(const rtengine::ProcEvent& event, c
+    Glib::ustring captionChanged, captionWriterChanged, headlineChanged, instructionsChanged,
+				  keywordAdded, keywordDeleted,
+				  categoryChanged, suppCategoryAdded, suppCategoryDeleted,
+				  creatorChanged, creatorJobTitleChanged, creditChanged, sourceChanged, copyrightChanged,
+				  cityChanged, provinceChanged, countryChanged,
+				  titleChanged, dateCreatedChanged, transReferenceChanged;
+	// flags are needed because just checking for an empty string would prevent replacing a tag with an empty string, i.e. clearing it
+	bool captionChangedFlag = false, captionWriterChangedFlag = false, headlineChangedFlag = false, instructionsChangedFlag = false,
+		 keywordAddedFlag = false, keywordDeletedFlag = false,
+		 categoryChangedFlag = false, suppCategoryAddedFlag = false, suppCategoryDeletedFlag = false,
+		 creatorChangedFlag = false, creatorJobTitleChangedFlag = false,
+		 creditChangedFlag = false, sourceChangedFlag = false, copyrightChangedFlag = false,
+		 cityChangedFlag = false, provinceChangedFlag = false, countryChangedFlag = false,
+		 titleChangedFlag = false, dateCreatedChangedFlag = false, transReferenceChangedFlag = false,
+		 tagFound;
+	 
+	if (event == rtengine::EvIPTC) {
+		// The following is necessary because it is not assured that initialPP[i].metadata.iptc is always initialized.
+		// There may be a better way to do this (existing functionality?) but I haven't found out how.
+		// Also, it might be considered to do this initialization in a more "general" place, e.g. procparams::PartialProfile constructor?
+		for (size_t i = 0; i < selected.size(); i++) {
+			if (initialPP[i].metadata.iptc.empty()) {
+				initialPP[i].metadata.iptc.insert(CAPTION, "");
+				initialPP[i].metadata.iptc.insert(CAPTION_WRITER, "");
+				initialPP[i].metadata.iptc.insert(CATEGORY, "");
+				initialPP[i].metadata.iptc.insert(CITY, "");
+				initialPP[i].metadata.iptc.insert(COPYRIGHT, "");
+				initialPP[i].metadata.iptc.insert(COUNTRY, "");
+				initialPP[i].metadata.iptc.insert(CREATOR, "");
+				initialPP[i].metadata.iptc.insert(CREATOR_JOB_TITLE, "");
+				initialPP[i].metadata.iptc.insert(CREDIT, "");
+				initialPP[i].metadata.iptc.insert(DATE_CREATED, "");
+				initialPP[i].metadata.iptc.insert(HEADLINE, "");
+				initialPP[i].metadata.iptc.insert(INSTRUCTIONS, "");
+				initialPP[i].metadata.iptc.insert(KEYWORDS, "");
+				initialPP[i].metadata.iptc[KEYWORDS].clear();
+				initialPP[i].metadata.iptc.insert(PROVINCE, "");
+				initialPP[i].metadata.iptc.insert(SOURCE, "");
+				initialPP[i].metadata.iptc.insert(SUPPLEMENTAL_CATEGORIES, "");
+				initialPP[i].metadata.iptc[SUPPLEMENTAL_CATEGORIES].clear();
+				initialPP[i].metadata.iptc.insert(TITLE, "");
+				initialPP[i].metadata.iptc.insert(TRANS_REFERENCE, "");
+			}
+		}
+		
+		// determine changes to IPTC tags of first selected picture
+		originalParams = initialPP[0];
+
+		if (pparams.metadata.iptc[CAPTION].at(0).compare(originalParams.metadata.iptc[CAPTION].at(0)) != 0) {
```

Comment:
This if-else-if chain belongs in `IPTCPanel::write`. The booleans like `captionChangedFlag` can be placed in `ParamsEdited`. The `Glib::ustring`s like `captionChanged` are not needed since they can be found in `pparams`. The exceptions are the added/deleted keyword/category, which can also go in `pparams`.